src/metal.zig

```diff
@@ -251,32 +262,118 @@ pub const Device = struct {
     }
 };
 
+pub const StreamingManager = struct {
+    const InflightBuffer = struct {
+        mtl_buffer: *mtl.Buffer,
+        queue: *Queue,
+        fence_value: u64,
+    };
+
+    mtl_device: *mtl.Device,
+    inflight_buffers: std.ArrayList(InflightBuffer),
+    free_buffers: std.ArrayList(*mtl.Buffer),
+
+    pub fn init(mtl_device: *mtl.Device) !*StreamingManager {
+        var manager = try allocator.create(StreamingManager);
```

Comment:
nit: I think elevating this decision (whether to heap allocate an object) to the caller usually leads to better memory management decisions